tests/performance/MemoryTest.cpp

```diff
@@ -345,21 +222,16 @@ TEST_P(MemoryTest, PrefetchPartitionsFromVersionedLayer) {
                          .WithMinLevel(level)
                          .WithTileKeys(tile_keys);
       total_requests_.fetch_add(1);
-      auto token = service_client.PrefetchTiles(
-          std::move(request),
-          [&](olp::dataservice::read::PrefetchTilesResponse response) {
-            if (response.IsSuccessful()) {
-              success_responses_.fetch_add(1);
-            } else {
-              failed_responses_.fetch_add(1);
-              ReportError(response.GetError());
-            }
-          });
+      auto result = service_client.PrefetchTiles(std::move(request));
 
-      RandomlyCancel(std::move(token));
+      auto response = result.GetFuture().get();
 
```

Comment:
Is RandomlyCancel not applicable for PrefetchPartitionsFromVersionedLayer?

Reply:
Reverted this change